src/service/action-impl.js

```diff
@@ -21,13 +21,13 @@ import {debounce, throttle} from '../utils/rate-limit';
 import {dev, user} from '../log';
 import {getMode} from '../mode';
 import {getValueForExpr} from '../json';
+import {hasOwn, map} from '../utils/object';
```

Comment:
Are we using `map`?

Reply:
Yes, but not for the whitelist. Would it be better if I create the whitelist with `map` in the service constructor to make it a prototypeless object?

Reply:
Oh I see you just moved it from below, nevermind. :)